### PR TITLE
Preference for Valentines

### PR DESCRIPTION
**Pull request by JamieD1**
Players can now opt out of Valentines events

:cl:  
rscadd: Opt out for Valentines
/:cl:

### CODE REVIEW

**Comment by patpol4** [comment]
Literally just reject the date, this is a nonissue imo.
Might lead to people disabling this and skipping the forced player interaction thus reducing rp.

**Comment by JamieD1** [comment]
"forced player interaction"

**Comment by patpol4** [comment]
hugbox

**Comment by Bibby0110** [comment]
Why the fuck is this only for donators?

**Comment by Tipy1802** [comment]
No fun allowed. The grinch is here to take away our holiday spirit because he has no valentine

**Comment by JamieD1** [comment]
Changed to All players

**Comment by bruhlookatthisdood** [comment]
> Changed to All players

now i can be lonley in peace

**Comment by Darkstick** [comment]
Hard no.
Forcing regulars out of their comfort zone is the fun part of the valentine's event.
You either deal with your valentine, or find something fun to do with your valentine. It does not matter if it disrupts your round, you should embrace unexpected happenings and find creative solutions, because that's what SS13 is all about.

**Comment by Xoxeyos** [comment]
I could go either way with this, but I believe AIs and Cyborgs shouldn't have the ability to obtain the valentine antag datum.
As per usual, the AI/cyborg must prioritize their laws over any secondary objective, or so I've been told. It really doesn't leave much room for the actual what may be brief, interactions with your valentine. You could also say that AIs/Cyborgs are unfeeling machines, but that doesn't really hold up for some players, so this last sentence is just food for thought.

All in all, it's hard to interact with your valentine as a silicon unit during a busy round, where we usually have a sustained pop of 40 to 50 players per day, while juggling your law set, and this "Protect "X", your valentine!".

**Comment by Moja10** [comment]
There is literally no rule or code that says you have to do anything different during valentine's event, you can tell your date to pound sand. Why make a preference for something already ignorable, other than being miserly and seeing redtext instills deep trauma

**Comment by optimumtact** [comment]
play 2 win

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -755,6 +755,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 						dat += "<b>Be [capitalize(i)]:</b> <a href='?_src_=prefs;preference=be_special;be_special_type=[i]'>[(i in be_special) ? "Enabled" : "Disabled"]</a><br>"
 			dat += "<br>"
 			dat += "<b>Midround Antagonist:</b> <a href='?_src_=prefs;preference=allow_midround_antag'>[(toggles & MIDROUND_ANTAG) ? "Enabled" : "Disabled"]</a><br>"
+			dat += "<b>Toggle Valentines Events:</b> <a href='?_src_=prefs;preference=donor;task=valentines'>[valentines ? "Yes" : "No"]</a><BR>"
```
Comment (Bibby0110) [inline code/modules/client/preferences.dm:758]: Ambiguous language.
Toggle it what? On? Off? Could be either with this wording

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -1251,6 +1252,8 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					donor_pda = (donor_pda % GLOB.donor_pdas.len) + 1
 				if("purrbation")
 					purrbation = !purrbation
+				if("valentines")
```
Comment (Bibby0110) [inline code/modules/client/preferences.dm:1255]: hidden inside is_donator block.

File: code/modules/events/holiday/vday.dm [hunk 3]
```diff
@@ -24,9 +24,10 @@
 	var/list/valentines = list()
 	for(var/mob/living/M in GLOB.player_list)
 		if(!M.stat && M.client && M.mind)
+			if(M.client.prefs.valentines == TRUE)
```
Comment (Bibby0110) [inline code/modules/events/holiday/vday.dm:27]:
```suggestion
			if(!M.client.prefs.valentines)
```
Assume FALSE = not participating

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
This should be merged. The ability to opt in or out optimizes which type of players are engaging in this RP concept, making the experience better for those who want RP. Furthermore, players should have the choice to play the game how they want to.

Moja puts it as “oh you can just tell them to pound sand”, but in this scenario the interaction ends there for the player who wants to RP.

Encouraging players to leave their comfort zone is horseshit. This is not how it works in reality. I believe people should be allowed to play as they want to so long as it is not interfering with the enjoyment of others. If anything, not having an opt out is detrimental to others enjoyment by reducing the likelihood of “good” RP.

I am always surprised by the urge of people who play quite infrequently to have players RP. The point of this game is to have fun. Fun does not necessarily mean you are engaging in roleplay, it can come in many forms. Furthermore, forcing players to engage in RP while still existing in a low to no rp environment is by no means enjoyable. We either go all the way with RP (which didn’t work, remember?) or stop these stupid antics and focus on bettering the experience of players by offering them a wide variety of ways to play the game.

To put it simply, you can’t have your cake and eat it too.

On a final note, this shouldn’t even be up to discussion. A button to decide whether or not to engage in an antagonist/irregular role is a familiar concept to SS13.

Forcing players to engage in anything is draconian and represents how militant some community members have gotten about their obsession with “roleplay”.

**Comment by optimumtact** [comment]
you sound like fun at parties.

**Comment by Moja10** [comment]
Being rejected from a date is RP

**Comment by Moja10** [comment]
You will be banned for not engaging in Valentines date RP

**Comment by Bibby0110** [comment]
> you sound like fun at parties.

spoiler: he isn't

**Comment by SomeguyManperson** [comment]
Wtf happened here

**Comment by ghost** [comment]
> Wtf happened here

just yogs tingz

**Comment by patpol4** [comment]
Just wanted to say that after giving it some thought i fully support this now. This is not a joke, i actually think this should be merged.

**Comment by Moja10** [comment]
There is literally no rule or code that says you have to do anything different during valentine's event, you can tell your date to pound sand. Why make a preference for something already ignorable, other than being miserly and seeing redtext instills deep trauma

**Comment by JamieD1** [comment]
> There is literally no rule or code that says you have to do anything different during valentine's event, you can tell your date to pound sand. Why make a preference for something already ignorable, other than being miserly and seeing redtext instills deep trauma

Because people are not emotionless twats

**Comment by MajManatee** [comment]
finally, legal erp

But yeah moja is right. Enable is kinda pointless. Just tell them to fuck off if you dont feel like bothering with them.

**Comment by MajManatee** [comment]
or you know, dont play the game for 24 hours. Believe it or not the world wont end.

**Comment by Moltijoe** [comment]
something like this being an option does seem preferrable

**Comment by optimumtact** [comment]
add this, but it transfers you to tgstation terry with all the other humans afraid of talking to other people

**Comment by patpol4** [comment]
#17682
